spring-test/src/main/java/org/springframework/test/web/servlet/result/MockMvcResultMatchers.java

```diff
@@ -204,6 +204,20 @@ public static JsonPathResultMatchers jsonPath(String expression, Object... args)
 		return new JsonPathResultMatchers(expression).value(matcher);
 	}
 
+	/**
+	 * An overloaded variant of {@link #jsonPath(String, Matcher)} (Matcher)} that also accepts
+	 * a target type for the resulting value that the matcher can work reliably against.
+	 * <p> This can be useful for matching numbers reliably &mdash; for example,
+	 * to coerce an integer into a double.</p>
+	 *
+	 * @param expression the JSON path expression
+	 * @param targetClass the target class to coerce the matching type into.
+	 * @param matcher a matcher for the value expected at the JSON path
+	 */
+	public static <T> ResultMatcher jsonPath(String expression, Class<T> targetClass, Matcher<T> matcher) {
+		return new JsonPathResultMatchers(expression).value(matcher, targetClass);
```

Comment:
Please use `targetType` instead of `targetClass`

Reply:
Already addressed locally